### PR TITLE
feat: [PL-23207]: with account id in few apis

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/10-common/components/Smtp/SmtpDetails.tsx [hunk 1]
```diff
@@ -46,7 +48,10 @@ const SmtpDetails: React.FC = () => {
   }
   const handleDelete = (): void => {
     if (data?.data?.uuid) {
-      deleteSmtp(data?.data?.uuid, { headers: { 'content-type': 'application/json' } })
+      deleteSmtp(data?.data?.uuid, {
+        headers: { 'content-type': 'application/json' },
+        queryParams: { accountIdentifier: accountId }
```
Comment (reetika04) [inline src/modules/10-common/components/Smtp/SmtpDetails.tsx:53]: Why are we repeating this?

Comment (mayankVermaHarness) [reply src/modules/10-common/components/Smtp/SmtpDetails.tsx:53]: +1

File: src/modules/20-rbac/modals/ApiKeyModal/views/ApiKeyForm.tsx [hunk 2]
```diff
@@ -46,21 +46,21 @@ const ApiKeyForm: React.FC<ApiKeyModalData> = ({ data, isEdit, onSubmit, apiKeyT
...
       if (isEdit) {
-        const updated = await editApiKey(values)
+        const updated = await editApiKey({ ...values, accountIdentifier: accountId })
         /* istanbul ignore else */ if (updated) {
           showSuccess(getString('rbac.apiKey.form.editSuccess', { name: values.name }))
           onSubmit?.(values)
         }
       } else {
-        const created = await createApiKey(values)
+        const created = await createApiKey({ ...values })
```
Comment (reetika04) [inline src/modules/20-rbac/modals/ApiKeyModal/views/ApiKeyForm.tsx:63]: Why?

**Review by reetika04 — APPROVED** [review]

File: src/modules/20-rbac/modals/ApiKeyModal/views/ApiKeyForm.tsx (hunk 2 above)
Comment (swarajcheguri-harness) [reply src/modules/20-rbac/modals/ApiKeyModal/views/ApiKeyForm.tsx:63]: added few data attributes before which needed spread, but later on removed old code.

**Review by mayankVermaHarness — APPROVED** [review]

**Review by knagurski — APPROVED** [review]
Apart from the unneeded spread (not a problem), it all looks good to me 👍🏻

**Pull request merged**